src/module.cc

```diff
@@ -0,0 +1,338 @@
+NAN_METHOD(TriggerReport) {
+  Nan::HandleScope scope;
+  v8::Isolate* isolate = Isolate::GetCurrent();
+  char filename[48] = "";
+
+  if (info[0]->IsString()) {
+    // Filename parameter supplied
+    Nan::Utf8String filename_parameter(info[0]->ToString());
+    if (filename_parameter.length() < 48) {
+      strcpy(filename, *filename_parameter);
+    } else {
+      Nan::ThrowSyntaxError("TriggerReport: filename parameter is too long (max 48 characters)");
+    }
+  }
+  if (info[0]->IsFunction()) {
+    // Callback parameter supplied
+    Nan::Callback callback(info[0].As<Function>());
+    // Creates a new Object on the V8 heap
+    Local<Object> obj = Object::New(isolate);
+    obj->Set(String::NewFromUtf8(isolate, "number"), Number::New(isolate, 54));
+    Local<Value> argv[1];
+    argv[0] = obj;
+    // Invoke the callback, passing the object in argv
+    callback.Call(1, argv);
+  }
+
+  TriggerNodeReport(isolate, kJavaScript, "JavaScript API", "TriggerReport (nodereport/src/module.cc)", filename);
+  // Return value is the NodeReport filename
+  info.GetReturnValue().Set(Nan::New(filename).ToLocalChecked());
+}
+
+/*******************************************************************************
+ * External JavaScript APIs for nodereport configuration
+ *
+ ******************************************************************************/
+NAN_METHOD(SetEvents) {
+  Nan::Utf8String parameter(info[0]->ToString());
+  v8::Isolate* isolate = node_isolate;
+  unsigned int previous_events = nodereport_events; // save previous settings
+  nodereport_events = ProcessNodeReportEvents(*parameter);
+
+  // If NodeReport newly requested for fatalerror, set up the V8 call-back
+  if ((nodereport_events & NR_FATALERROR) && !(previous_events & NR_FATALERROR)) {
+    isolate->SetFatalErrorHandler(OnFatalError);
+  }
+
+  // If NodeReport newly requested for exceptions, tell V8 to capture stack trace and set up the callback
+  if ((nodereport_events & NR_EXCEPTION) && !(previous_events & NR_EXCEPTION)) {
+    isolate->SetCaptureStackTraceForUncaughtExceptions(true, 32, v8::StackTrace::kDetailed);
+    // The hook for uncaught exception won't get called unless the --abort_on_uncaught_exception option is set
+    v8::V8::SetFlagsFromString("--abort_on_uncaught_exception", sizeof("--abort_on_uncaught_exception")-1);
+    isolate->SetAbortOnUncaughtExceptionCallback(OnUncaughtException);
+  }
+
+#ifndef _WIN32
+  // If NodeReport newly requested on external user signal set up watchdog thread and callbacks
+  if ((nodereport_events & NR_SIGNAL) && !(previous_events & NR_SIGNAL)) {
+    uv_sem_init(&report_semaphore, 0);
+    if (StartWatchdogThread(ReportSignalThreadMain) == 0) {
+      uv_async_init(uv_default_loop(), &nodereport_trigger_async, SignalDumpAsyncCallback);
+      uv_unref(reinterpret_cast<uv_handle_t*>(&nodereport_trigger_async));
+      RegisterSignalHandler(nodereport_signal, SignalDump, false);
+    }
+  }

```

Comment:
You should check the return values of uv_sem_init() and uv_async_init() because those can fail (and sometimes will, i.e., it's not academic.)